xdsl/utils/mlir_lexer.py

```diff
@@ -64,6 +64,10 @@ def string_contents(self):
 
     @property
     def bytes_contents(self) -> bytes:
+        # NOTE: Early return if no escape sequences are present.
+        if "\\" not in self.text[1:-1]:
+            return self.text[1:-1].encode()
+
```

Comment:
Would it make sense to do something like split the string on escape characters instead of special casing?

Reply:
Probably, so scanning for the next index of `\` and inspecting the next char, instead of iterating over each character? That seems like a better approach anyway.

Reply:
`"\\" == r"\"` right? so should just be collect up to the next `\` and encode that, rather than doing 1 character at a time (which I assume but don't know is the slow bit.

Reply:
@alexarice @superlopuh That sounds good! I will try this approach these days.